Sources/Honeycomb/HoneycombNavigationSpanInstrumentation.swift

```diff
@@ -16,76 +21,158 @@ func getTracer() -> Tracer {
 internal class HoneycombNavigationProcessor {
     static let shared = HoneycombNavigationProcessor()
     var currentNavigationPath: [String] = []
+    var lastNavigationTime: Date? = nil
 
-    private init() {}
+    private init() {
+        setupAppLifecycleTracking()
+    }
 
     @available(iOS 16.0, macOS 13.0, *)
-    func reportNavigation(path: NavigationPath) {
+    func reportNavigation(path: NavigationPath, reason: String? = nil) {
         if let codablePath = path.codable {
-            reportNavigation(path: codablePath)
+            reportNavigation(path: codablePath, reason: reason)
         } else {
-            reportNavigation(path: unencodablePath)
+            reportNavigation(path: unencodablePath, reason: reason)
         }
     }
 
-    func reportNavigation(path: String) {
-        currentNavigationPath = [path]
-
-        // emit a span that says we've navigated to this path
-        getTracer().spanBuilder(spanName: navigationSpanName)
-            .setAttribute(key: "screen.name", value: path)
-            .startSpan()
-            .end()
+    func reportNavigation(path: String, reason: String? = nil) {
+        reportNavigation(path: [path], reason: reason)
     }
 
-    func reportNavigation(path: Encodable) {
+    func reportNavigation(path: Encodable, reason: String? = nil) {
         do {
             let encoder = JSONEncoder()
             let data = try encoder.encode(path)
             let pathStr = String(decoding: data, as: UTF8.self)
 
-            reportNavigation(path: pathStr)
+            reportNavigation(path: pathStr, reason: reason)
         } catch {
-            reportNavigation(path: unencodablePath)
+            reportNavigation(path: unencodablePath, reason: reason)
         }
     }
 
-    func reportNavigation(path: [Encodable]) {
+    func reportNavigation(path: [String], reason: String? = nil) {
+        // If we have a previous navigation, emit a "NavigationFrom" span
+        navigationEnd(reason: reason ?? "navigation")
+
+        // Update current path
+        currentNavigationPath = path
+
+        navigationStart(reason: reason ?? "navigation")
+    }
```

Comment:
this change is subtle, might not pop up right away (and we didn't have tests exercising this) 

the navigation split view in the examples builds an explicit path, ex

```swift
let path: [Encodable] = ["Split View", park, tree]
Honeycomb.setCurrentScreen(path: path)
```

and previously this would hit the function definition below (which accepts a `[Encodable]`). That would get stringified into a single long string and get used as both the path _and_ the current screen name 

instead, now we preserve the array that the client gave us as the full path and just use the last entry as the current screen name